### PR TITLE
miru: init at 4.5.10

### PR DESCRIPTION
**Pull request by d4ilyrun**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Please rebase

**Comment by abathur** [comment]
@d4ilyrun Sorry this has languished without getting a proper review.

We can try to get it moving, but unfortunately I'll have to ask you to jump through a few more hoops...:
- You'll need to rebase against nixpkgs master again. It looks like both pkgs/top-level/all-packages.nix and maintainers/maintainer-list.nix have merge conflicts to fix.
- Please update the PR title to match the new version target (it still indicates 2.2.2, but the commit indicates 3.10.6).
- The CI report from your last push indicates at least 2 problems:
    - Your maintainer entry isn't in the right order alphabetically. It looks like you appended it to the end, but it'll need to go in the right spot. At the time I'm writing this, I imagine that means it needs to go between `cyplo` and `dadada`.
    - When you updated the PR to target version 3.10.6, you didn't update the hash/sha256. You'll need to update it to match the new source (and make sure the project still builds/works without needing additional changes).
    
        (You likely didn't notice this problem on your end since you'd already built the 2.2.2 version on your system. Nix uses this hash to look for an existing copy in the store--it won't rebuild if you change the version but leave the hash alone. If you empty the hash string and try to build it, Nix will pull the source again and let you know what the new hash is.)
        
- @onny marked your PR as a draft when he asked you to rebase, which lets reviewers know that the PR needs some work or changes before it's reviewable. Once you make the changes on my list and re-push, please also mark it as ready for review!

(Fair warning: I'm just asking for the things I'm pretty sure you'll need to get this ready for review. Reviewers will likely request at least one more round of changes, if not a few.)

**Comment by abathur** [comment]
I noticed there's another draft PR open to add this package. If you decide to put additional effort into this one, check up on its progress first:
- #238429

If it bogs down, you can also use the review feedback it received as a reference to help handle some issues before review.

**Comment by d4ilyrun** [comment]
Hello @abathur ! 
Sorry I took so long, I just rebased my branch and followed your advice, it should be good for another review now 😄

**Comment by onemoresuza** [comment]
There's a new convention for packages path: [by-name](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md).

In this case, `pkgs/applications/video/miru/default.nix` should be moved to `pkgs/by-name/mi/miru/package.nix`.

**Comment by abathur** [comment]
I'm not a decider on this and don't know if we've concluded all new packages _must_ follow the by-name convention--and it's moot if the author eagerly adopts the convention--but I do want to ask anyone making a review/merge call on this point to weigh the risk of bogging down this long-deferred PR over something that wasn't a norm when it was opened and can almost certainly be handled during the eventual auto-migration.

**Comment by onemoresuza** [comment]
Fair point, @abathur.

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 172872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>miru</li>
  </ul>
</details>

- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] all depending packages build
- [X] `AppImage` application states its `sourceProvenance` as `binaryNativeCode`

# Possible improvements

- The `mainProgram` meta attribute does not match the binary under `$out/bin`:
  instead of `miru`, there's a `miru-4.5.5`. An explicit copy of the
  binary to `$out/bin/miru` should fix it.

**Comment by d4ilyrun** [comment]
Oh sorry I did not see there was a new way of adding packages.
 I updated everything to follow the new convention, and renamed the resulting binary's name.

**Review by onemoresuza — APPROVED** [review]
Good job, @d4ilyrun. Just remember that before a commiter may merge this PR, the commits
ought to be squashed.

File: pkgs/by-name/mi/miru/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/mi/miru/package.nix:9]: We no longer need to generate a name variable since all tools accept pname version.

**Comment by heisfer** [comment]
I think most of the stuff can be copy pasta from here https://github.com/heisfer/nixdot/blob/main/pkgs/miru/default.nix

File: pkgs/by-name/mi/miru/package.nix
Comment (d4ilyrun) [reply pkgs/by-name/mi/miru/package.nix]: I redefined it since a similar pattern appeared inside the url and install commands.  
Just removed it and squashed everything.

**Review by onemoresuza — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/mi/miru/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/mi/miru/package.nix:43]:
```suggestion
      Completely ad free with no tracking/data collection.
```

File: pkgs/by-name/mi/miru/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/mi/miru/package.nix:53]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Besides, LFTM

**Pull request merged**
